### PR TITLE
LPS-101084 Apply view count service to AssetEntry

### PR DESCRIPTION
**Pull request by stsquared99**

### CODE REVIEW

**Comment by stsquared99** [comment]
ci:test

**Comment by vicnate5** [comment]
ci:test

**Comment by stsquared99** [comment]
ci:test

**Comment by shuyangzhou** [comment]
@stsquared99 this does not contain Lily's fix to you, right? Please add it in and retest

**Comment by stsquared99** [comment]
ci:stop

**Comment by stsquared99** [comment]
ci:test

**Comment by vicnate5** [comment]
@stsquared99 Still getting the post-upgrade startup race condition
https://testray.liferay.com/reports/production/logs/test-1-14/1571857720451/test-portal-acceptance-pullrequest(master)/4996/functional-upgrade-tomcat90-mysql57-jdk8/3/0/liferay-log.txt.gz

There is also more test failures on DB2 on this PR than at head, specifically the search tests
https://test-1-23.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/197598/#showFailuresLink

**Comment by stsquared99** [comment]
ci:test

**Comment by vicnate5** [comment]
@stsquared99 post-upgrade startup issue again here:
https://testray.liferay.com/reports/production/logs/test-1-23/1571957194860/test-portal-acceptance-pullrequest(master)/4825/functional-upgrade-tomcat90-db2111-jdk8/3/0/jenkins-console.txt.gz
See also 
https://testray.liferay.com/reports/production/logs/test-1-23/1571957194860/test-portal-acceptance-pullrequest(master)/4825/functional-upgrade-tomcat90-postgresql10-jdk8/3/0/liferay-log.txt.gz
https://testray.liferay.com/reports/production/logs/test-1-23/1571957194860/test-portal-acceptance-pullrequest(master)/4825/functional-upgrade-tomcat90-mysql57-jdk8/3/0/liferay-log.txt.gz

**Comment by stsquared99** [comment]
@vicnate5 FYI this next test run is just to get a stacktrace

**Comment by stsquared99** [comment]
ci:test

**Comment by shuyangzhou** [comment]
@stsquared99 how is this going on?

**Comment by stsquared99** [comment]
ci:test

**Comment by stsquared99** [comment]
ci:test

**Comment by Preston-Crary** [comment]
@stsquared99 my best guess is that the column name in the alter statement cares about the case, try using DBInspector.normalizeName() on line 52 of UpgradeViewCount.

**Comment by stsquared99** [comment]
ci:test

**Comment by stsquared99** [comment]
ci:test

**Comment by stsquared99** [comment]
ci:stop

**Comment by stsquared99** [comment]
ci:test

**Comment by stsquared99** [comment]
ci:stop

**Comment by stsquared99** [comment]
ci:test

**Comment by stsquared99** [comment]
@shuyangzhou @Preston-Crary No unique failures on this last run

**Comment by stsquared99** [comment]
ci:test:upgrade

**Comment by stsquared99** [comment]
ci:test:upgrade

**Comment by shuyangzhou** [comment]
@vicnate5 please confirm, thanks!

**Comment by vicnate5** [comment]
@shuyangzhou I helped @stsquared99 review the test results, this is good.

File: portal-impl/src/com/liferay/portlet/asset/service/impl/AssetEntryLocalServiceImpl.java [hunk 1]
```diff
@@ -450,12 +455,11 @@ public AssetEntry incrementViewCounter(
...
 
-	@BufferedIncrement(
-		configuration = "AssetEntry", incrementClass = NumberIncrement.class
-	)
 	@Override
+	@Transactional(enabled = false)
 	public void incrementViewCounter(
-		long userId, String className, long classPK, int increment) {
+		long companyId, long userId, String className, long classPK,
+		int increment) {
```
Comment (shuyangzhou) [inline portal-impl/src/com/liferay/portlet/asset/service/impl/AssetEntryLocalServiceImpl.java]: @stsquared99 this "increment" parameter is not used at all.
This convenient method https://github.com/shuyangzhou/liferay-portal/pull/8416/files#diff-5607c7ad65ea9f759fa3f371e303a802R29 should be removed. When not having it, it won't be possible to make mistakes like this.

Please remove that method regenerate, cleanup exist usages and resend.

**Comment by shuyangzhou** [comment]
@stsquared99 see my comment above, fix it and resend, thanks!

**Comment by stsquared99** [comment]
Resent https://github.com/shuyangzhou/liferay-portal/pull/8466

**Pull request closed without merging**